Add hover text

bkoropoff opened the pull request:
If the link text is not literally the full identifier, show the full identifier on hover.

This will be handy for the way I intend to handle https://github.com/mkdocstrings/mkdocstrings/issues/269

oprypin commented:
Hmm so this change is good for mkdocstrings usage but I think generally this effect can be undesirable. Consider the main example:

https://github.com/mkdocstrings/autorefs#usage

Or um
if I write
```markdown
Link to [Hello, World!][hello-world]
```
I don't want to have the popup text "hello-world", what's the use of it?

So I think it should be mkdocstrings itself that somehow indicates that a hover title should be used.

bkoropoff commented:
@oprypin I changed it to use a different reference type.  I noticed `mkdocstrings-identifier` seemed to be unused, so I removed it from the regex as well.

oprypin commented:
Thanks.
Hmm I don't understand something.
If currently mkdocstrings produces only "required references", shouldn't the new thing be "required references with title"? Instead you seem to be adding "optional references with title".

bkoropoff commented:
https://github.com/mkdocstrings/mkdocstrings/issues/269 uses this to automatically cross-reference base classes.  If the base class isn't in the same package, we don't want to warn about an unmapped reference.
